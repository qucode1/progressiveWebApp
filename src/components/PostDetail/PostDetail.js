import React from 'react'
import Post from '../Post/Post'

import posts from '../../blog-posts.json'

const PostDetail = function (props, { match }) {
  const matchingPost = posts.posts.find((post) => post.slug === props.match.params.slug)
  return (
    <div>
      <Post {...matchingPost} details />
    </div>
  )
}
export default PostDetail
