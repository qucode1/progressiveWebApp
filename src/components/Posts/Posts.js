import React from 'react'
import { Route, Switch, Link } from 'react-router-dom'

import Post from '../Post/Post'
import posts from '../../blog-posts.json'
import PostDetail from '../PostDetail/PostDetail'

import styles from '../Post/Post.css'

const PostList = () => (
  <div>
    {posts.posts.map((post) => (
      <Post key={post.slug} {...post} details={false} />
    ))}
    <ul>
      <li>
        <Link to='/posts/test'><h2 className={styles.title}>Using TravisCI (will not work)</h2></Link>
      </li>
      <li>
        <Link to='/test'>Go Somewhere (Won't work)</Link>
      </li>
    </ul>
  </div>
)
// PostDetail needs a 'wrapper component' to pass the posts data as a prop,
// without losing the match object
const PostDetailData = (props) => (
  <PostDetail posts={posts} match={props.match} />
)

const Posts = function ({ match }) {
  return (
    <div>
      <Switch>
        <Route path={`${match.url}/:slug`} component={PostDetailData} />
        <Route path={`${match.url}`} component={PostList} />
      </Switch>
    </div>
  )
}
export default Posts
