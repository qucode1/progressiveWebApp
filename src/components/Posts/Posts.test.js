import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import Posts from './Posts'
import PostDetail from '../PostDetail/PostDetail'

const posts = {
  'posts': [{
    'slug': 'post-slug'
  }]
}

const match = {
  'url': '/posts',
  'params': {
    'slug': 'post-slug'
  }
}

test('Should render PostDetails when visiting /posts/:slug', () => {
  const component = mount(
    <MemoryRouter initialEntries={['/posts/testing-with-jest-and-enzyme']} initialIndex={0} >
      <Posts match={match}/>
    </MemoryRouter>
  )

  expect(component.find(PostDetail).length).toBe(1)
})

test('Should render Post List', () => {
  const component = mount(
    <MemoryRouter initialEntries={['/posts']} initialIndex={0} >
      <Posts match={match}/>
    </MemoryRouter>
  )

  expect(component.find('ul').length).toBe(1)
})

it('Posts Component should render as expected', () => {
  const wrapper = shallow(<Posts match={match}/>)
  const tree = toJson(wrapper)
  expect(tree).toMatchSnapshot()
})
