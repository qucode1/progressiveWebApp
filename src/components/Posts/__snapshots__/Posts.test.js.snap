// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component should render as expected 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      path="/posts/:slug"
    />
    <Route
      component={[Function]}
      path="/posts"
    />
  </Switch>
</div>
`;
